packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -24,33 +24,87 @@ export const displayTxResult = (
   if (typeof displayContent === "bigint") {
     try {
       const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
+      if (asText) {
```

Comment:
Looks like all the logic inside this `if` is already in `NumberDisplay` component, so probably no need to duplicate it here?

Reply:
Ohh yeah, we could safely this condition and also `asText` param from `displayContent` 

It was only used to handle the elements inside array's,  but since now we are nicely rendering each of elements with component we could remove this. 

So yup lets remove this

Reply:
Should be updated in commit https://github.com/ByteAtATime/scaffold-eth-2/commit/946900a6747e508d46a9d614450618c56c9a7c9f, I'd be grateful if someone can check this